jina/peapods/runtimes/asyncio/rest/models.py

```diff
@@ -1,11 +1,153 @@
-from typing import Callable, Dict, Any, Optional, List, Union
+from typing import Dict, Any, Optional, List, Union
+
+from enum import Enum
+from datetime import datetime
+from collections import defaultdict
 
 from pydantic import Field, BaseModel, create_model
+from google.protobuf.descriptor import Descriptor, FieldDescriptor
+from google.protobuf.pyext.cpp_message import GeneratedProtocolMessageType
 
 from jina.enums import DataInputType
 from jina.types.document import Document
 from jina.parsers import set_client_cli_parser
-from jina.proto.jina_pb2 import DocumentProto, QueryLangProto
+from jina.proto.jina_pb2 import (
+    DenseNdArrayProto,
+    NdArrayProto,
+    SparseNdArrayProto,
+    NamedScoreProto,
+    DocumentProto,
+    RouteProto,
+    EnvelopeProto,
+    StatusProto,
+    MessageProto,
+    RequestProto,
+    QueryLangProto,
+)
+
+
+DEFAULT_REQUEST_SIZE = set_client_cli_parser().parse_args([]).request_size
+PROTO_TO_PYDANTIC_MODELS = {}
+PROTOBUF_TO_PYTHON_TYPE = {
+    FieldDescriptor.TYPE_INT32: int,
+    FieldDescriptor.TYPE_INT64: int,
+    FieldDescriptor.TYPE_UINT32: int,
+    FieldDescriptor.TYPE_UINT64: int,
+    FieldDescriptor.TYPE_SINT32: int,
+    FieldDescriptor.TYPE_SINT64: int,
+    FieldDescriptor.TYPE_BOOL: bool,
+    FieldDescriptor.TYPE_FLOAT: float,
+    FieldDescriptor.TYPE_DOUBLE: float,
+    FieldDescriptor.TYPE_FIXED32: float,
+    FieldDescriptor.TYPE_FIXED64: float,
+    FieldDescriptor.TYPE_SFIXED32: float,
+    FieldDescriptor.TYPE_SFIXED64: float,
+    FieldDescriptor.TYPE_BYTES: bytes,
+    FieldDescriptor.TYPE_STRING: str,
+    FieldDescriptor.TYPE_ENUM: Enum,
+    FieldDescriptor.TYPE_MESSAGE: None,
+}
+
+
+def protobuf_to_pydantic_model(
+    protobuf_model: Union[Descriptor, GeneratedProtocolMessageType]
+) -> BaseModel:
+    """
+    Converts Protobuf messages to Pydantic model for jsonschema creation/validattion
+
+    ..note:: Model gets assigned in the global dict :data:PROTO_TO_PYDANTIC_MODELS
+
+    :param protobuf_model: *Proto message from proto file
+    :type protobuf_model: Union[Descriptor, GeneratedProtocolMessageType]
+    :return: Pydantic model
+    :rtype: BaseModel
+    """
+
+    all_fields = {}
+    oneof_fields = defaultdict(list)
+
+    if isinstance(protobuf_model, Descriptor):
+        model_name = protobuf_model.name
+        protobuf_fields = protobuf_model.fields
+    elif isinstance(protobuf_model, GeneratedProtocolMessageType):
+        model_name = protobuf_model.DESCRIPTOR.name
+        protobuf_fields = protobuf_model.DESCRIPTOR.fields
+
+    if model_name.endswith('Proto') and model_name in PROTO_TO_PYDANTIC_MODELS:
```

Comment:
this assumption is needed? isnt enough to remove this endswith? I think it adds a big assumption for the future